client/gutenberg/extensions/subscriptions/edit.jsx

```diff
@@ -0,0 +1,72 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { Component } from '@wordpress/element';
+import { TextControl, Button, ToggleControl } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { __ } from 'gutenberg/extensions/presets/jetpack/utils/i18n';
+import { sprintf } from '@wordpress/i18n/build/index';
+import apiFetch from '@wordpress/api-fetch';
+
+class SubscriptionEdit extends Component {
+	componentDidMount() {
+		// Get the subscriber count so it is available right away if the user toggles the setting
+		this.get_subscriber_count();
+	}
+
+	render() {
+		const { attributes, className, isSelected, setAttributes } = this.props;
+		const { subscribePlaceholder, showSubscribersTotal, subscriberCountString } = attributes;
+
+		if ( isSelected ) {
+			return (
+				<div className={ className } role="form">
+					<ToggleControl
+						label={ __( 'Show total subscribers' ) }
+						checked={ showSubscribersTotal }
+						onChange={ () => {
+							setAttributes( { showSubscribersTotal: ! showSubscribersTotal } );
+						} }
+					/>
+					<TextControl placeholder={ subscribePlaceholder } required onChange={ () => {} } />
+					<Button type="button" isDefault>
+						{ __( 'Subscribe' ) }
+					</Button>
+				</div>
+			);
+		}
+
+		return (
+			<div className={ className } role="form">
+				{ showSubscribersTotal && <p role="heading">{ subscriberCountString }</p> }
+				<TextControl placeholder={ subscribePlaceholder } />
+				<Button type="button" isDefault>
+					{ __( 'Subscribe' ) }
+				</Button>
+			</div>
+		);
+	}
+
+	get_subscriber_count() {
+		const { setAttributes } = this.props;
+
+		apiFetch( { path: '/wpcom/v2/subscribers/count' } ).then( count => {
```

Comment:
Maybe not for this PR but noting in case — should this handle situations where API returns 500 (or even 404?). Or missing or invalid data?

Gutenberg's [`apiFetch`](https://github.com/WordPress/gutenberg/tree/master/packages/api-fetch#readme) is just a wrapper for JS `fetch` which has its quirks when dealing with HTTP errors; they all end up in `.then` block instead of `.catch`: https://www.tjvantoll.com/2015/09/13/fetch-and-errors/